pydicom/dataelem.py

```diff
@@ -212,7 +212,7 @@ def from_json(cls, dataset_class, tag, vr, value, value_key,
                 raise TypeError(fmt.format(value_key, tag))
         elif value_key in {'InlineBinary', 'BulkDataURI'}:
             if isinstance(value, list):
-                fmt = '"{}" of data element "{}" must be a string.'
+                fmt = '"{}" of data element "{}" must be a bytes-like object.'
```

Comment:
Seems like a URI would not be expected to be a bytes object, but rather a string?  I had a quick look for tests on the URI pathway, but didn't see any

Reply:
Yes, you are right - I have only thought about the `InlineBinary`, as that caused the problem, not about `BulkDataURI`. Have to re-phrase this, or just use different messages for these options.

Reply:
As for tests - we definitely need more here; maybe I will look at this sometime later.